bundle-workflow/python/test_workflow/local_test_cluster.py

```diff
@@ -51,3 +58,37 @@ def destroy(self):
             self.stdout.close()
             self.stderr.close()
             self.process = None
+
+    def url(self, path=''):
+        return f'{"https" if self.security_enabled else "http"}://{self.endpoint()}:{self.port()}{path}'
+
+    def download(self):
+        print(f'Creating local test cluster in {self.work_dir}')
+        os.chdir(self.work_dir)
+        print(f'Downloading bundle from {self.manifest.build.location}')
+        urllib.request.urlretrieve(self.manifest.build.location, 'bundle.tgz')
+        print(f'Downloaded bundle to {os.path.realpath("bundle.tgz")}')
+
+        print('Unpacking')
+        subprocess.check_call('tar -xzf bundle.tgz', shell = True)
+        print('Unpacked')
+
+    def disable_security(self, dir):
+        subprocess.check_call(f'echo "plugins.security.disabled: true" >> {os.path.join(dir, "config", "opensearch.yml")}', shell = True)
+
+    def wait_for_service(self):
+        print('Waiting for service to become available')
+        url = self.url('/_cluster/health')
+
+        for attempt in range(10):
+            try:
+                print(f'Pinging {url} attempt {attempt}')
+                response = requests.get(url, verify = False, auth = ('admin', 'admin'))
```

Comment:
Passing basic auth information for the non-security case is OK because the server just ignores it. Yes, I know it's sending a username and password over an unencrypted channel, but the default username and password are hardly secrets - they're right there in the code!